ts/Core/Series/Series.ts

```diff
@@ -4339,22 +4353,21 @@ class Series {
      */
     public hasOptionChanged(optionName: string): boolean {
         const chart = this.chart,
-            option = (this.options as any)[optionName],
+            option = this.options[optionName as keyof SeriesOptions],
             plotOptions = chart.options.plotOptions,
-            oldOption = (this.userOptions as any)[optionName];
+            oldOption = this.userOptions[optionName as keyof DeepPartial<SeriesOptions>],
+            plotOptionsOption = pick(
+                plotOptions?.[this.type]?.[optionName as keyof Omit<SeriesOptions, NonPlotOptions>],
+                plotOptions?.series?.[optionName as keyof Omit<SeriesOptions, NonPlotOptions>]
+            );
 
-        if (oldOption) {
+        if (oldOption && !defined(plotOptionsOption)) {
```

Comment:
It would be good to add a comment here with the issue number and a short description of why it is here.